### PR TITLE
Added Gles2 warning popup

### LINKED ISSUE #1565: Add warning popup for GLES2 mode
When the game starts in GLES2 mode a warning popup should be shown once entering the main menu, because we don't test a lot and it has at least one serious issue (and some different lighting): https://github.com/Revolutionary-Games/Thrive/issues/1564

### PR DESCRIPTION
**Pull request by icedjuro**
Closes #1565

### CODE REVIEW

File: src/gui_common/MainMenu.cs [hunk 1]
```diff
@@ -133,6 +138,9 @@ private void RunMenuSetup()
         // Easter egg message
         ToolTipHelper.RegisterToolTipForControl(
             thriveLogo, toolTipCallbacks, ToolTipManager.Instance.GetToolTip("thriveLogoEasterEgg", "mainMenu"));
+
+        if ((int)OS.GetCurrentVideoDriver() == 1)
```
Comment (hhyyrylainen) [inline src/gui_common/MainMenu.cs:142]: Don't cast to int, compare the enum values directly.

Comment (hhyyrylainen) [reply src/gui_common/MainMenu.cs:142]: Also this code shouldn't retrigger when returning from the game so also check `IsReturningToMenu` here.

File: src/gui_common/MainMenu.tscn [hunk 2]
```diff
@@ -324,6 +325,17 @@ __meta__ = {
 
 [node name="GUIAnimations" type="AnimationPlayer" parent="."]
 anims/MenuSlide = SubResource( 1 )
+
+[node name="GLES2 Popup" type="ConfirmationDialog" parent="."]
+margin_right = 515.0
+margin_bottom = 108.0
+theme = ExtResource( 2 )
+window_title = "GLES2 Mode Warning"
+dialog_text = "You are running Thrive with GLES2. This is severely untested and is likely to cause issues."
```
Comment (hhyyrylainen) [inline src/gui_common/MainMenu.tscn:334]: Advice on trying to update video drivers and force nvidia or amd graphics to be used, would be nice to include here.

**Review by hhyyrylainen — CHANGES_REQUESTED** [review]

File: src/gui_common/MainMenu.tscn [hunk 3]
```diff
@@ -324,6 +324,17 @@ __meta__ = {
 
 [node name="GUIAnimations" type="AnimationPlayer" parent="."]
 anims/MenuSlide = SubResource( 1 )
+
+[node name="GLES2 Popup" type="ConfirmationDialog" parent="."]
+margin_right = 515.0
+margin_bottom = 108.0
+theme = ExtResource( 2 )
+window_title = "GLES2 Mode Warning"
+dialog_text = "You are running Thrive with GLES2. This is severely untested and is likely to cause issues. Try to update your video drivers or force AMD/Nvidia graphics to be used."
```
Comment (hhyyrylainen) [inline src/gui_common/MainMenu.tscn:333]: `Try to update your video drivers and/or force AMD or Nvidia graphics to be used to run Thrive.`
I think that would be a bit better form to say basically the same thing.

**Review by hhyyrylainen — CHANGES_REQUESTED** [review]

File: src/gui_common/MainMenu.cs [hunk 4]
```diff
@@ -29,9 +29,11 @@ public class MainMenu : Node
     [Export]
     public NodePath FreebuildButtonPath;
 
+    [Export]
+    public NodePath GLES2PopupPath;
```
Comment (hhyyrylainen) [inline src/gui_common/MainMenu.cs:33]: This doesn't run for me. I think you forgot to include the change to include this in the main menu scene

File: src/gui_common/MainMenu.cs [hunk 5]
```diff
@@ -29,9 +29,11 @@ public class MainMenu : Node
...
+
     public Array MenuArray;
     public TextureRect Background;
-
```
Comment (hhyyrylainen) [inline src/gui_common/MainMenu.cs:34]: Please put this blank line back in, I think it's nice to separate logically different fields.

**Review by hhyyrylainen — CHANGES_REQUESTED** [review]

File: src/gui_common/MainMenu.cs (hunk 4 above)
Comment (icedjuro) [reply src/gui_common/MainMenu.cs:33]: nodepath got messed up, will add back

File: src/gui_common/MainMenu.tscn (hunk 2 above)
Comment (hhyyrylainen) [inline src/gui_common/MainMenu.tscn:334]: I think you accidentally deleted the rest of the message?

File: src/gui_common/MainMenu.tscn [hunk 6]
```diff
@@ -324,6 +325,17 @@ __meta__ = {
 
 [node name="GUIAnimations" type="AnimationPlayer" parent="."]
 anims/MenuSlide = SubResource( 1 )
+
+[node name="GLES2 Popup" type="ConfirmationDialog" parent="."]
```
Comment (hhyyrylainen) [inline src/gui_common/MainMenu.tscn:329]: Please change the type here to be a box with just the ok button as it seems the cancel / ok button don't do anything different.

**Review by hhyyrylainen — CHANGES_REQUESTED** [review]

**Review by hhyyrylainen — APPROVED** [review]

**Pull request merged**